src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,132 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+export class TodoList extends React.Component {
+  state = {
+    query: '',
+    statusFilter: 'all',
+  };
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
+  handleChange = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({ [name]: value });
+  };
+
+  filterByQuery = () => {
+    const { query } = this.state;
+
+    return (task) => {
+      if (query) {
+        return (
+          task.title !== null
+          && task.title.toLowerCase().includes(query.toLowerCase())
+        );
+      }
+
+      return task;
+    };
+  }
+
+  filterByStatus = () => {
+    const { statusFilter } = this.state;
+
+    return (task) => {
+      if (statusFilter === 'Active') {
+        return task.completed === false;
+      }
+
+      if (statusFilter === 'Completed') {
+        return task.completed === true;
+      }
+
+      return task;
+    };
+  }
```

Comment:
let's use switch case

```suggestion
  filterByStatus = (todo) => {
    const { statusFilter } = this.state;

      if (statusFilter === 'Active') {
        return task.completed;
      }

      if (statusFilter === 'Completed') {
        return task.completed;
      }

      return task;
  }
```